### PR TITLE
[AIRFLOW-5440] Static checks from docker file use dumb-init

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5440

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This is needed so that you can easily kill such checks with ^C
Not doing it might cause your docker containers run for a long time
and take precious resources.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
CC: @nuclearpinguin -> this was the most likely culprit from what you saw as "hanging" mypy. The problem was that when you thought you killed a running pre-commit check with ^C, the running container were still running. Pre-commit runs as many containers in the background as many processes you have so if you start running one or two of those in parallel  and a docker build on top of it, you can imagine that docker might get totally stuck (especially on MacOS where we mount files over the super-slow - in some cases - osxfs). In this case, it gets docker so slow that it even stops responding for a long time to a docker ps command from what I saw.

You can read more about it in here, where docker developers explain performance issues on osx and that they are working on improving it: https://docs.docker.com/docker-for-mac/osxfs/#performance-issues-solutions-and-roadmap

This commit addresses the problem in the way that any cancelled pre-commit or script-originated static check (with Ctrl^C/SIGTERM or any other catchable signal) will also cancel running pre-commit checks. Unlike running the The dumb-init entrypoint is designed in the way to propagate such signals to scripts executed as parameter of the entrypoint. I tested it and could reproduce both behaviours - before and after dumb-init and it works as expected (before dumb-init, cancelling pre-commit did not kill the running containers, after dumb-init they are killed immediately). You can read more about the problem here: https://github.com/Yelp/dumb-init

File: scripts/ci/_utils.sh [hunk 1]
```diff
@@ -610,102 +611,114 @@ function run_flake8() {
...
-            "${AIRFLOW_SLIM_CI_IMAGE}"
+            "${AIRFLOW_SLIM_CI_IMAGE}" \
+            "--" "/opt/airflow/docs/build.sh" \
+            | tee -a "${OUTPUT_LOG}"
 }
 
 function run_check_license() {
     docker run "${AIRFLOW_CONTAINER_EXTRA_DOCKER_FLAGS[@]}" -t \
-            --entrypoint /opt/airflow/scripts/ci/in_container/run_check_licence.sh \
+            --entrypoint "/usr/bin/dumb-init"  \
```
Comment (potiuk) [inline scripts/ci/_utils.sh:655]: NOTE: Just a comment - this is CORRECT. dumb-init installed in CheckLicence image uses different installation method and it actually lands in /usr/bin not /usr/local/bin as the dumb-init installed via pip (in CI/CI-slim images).

**Comment by potiuk** [comment]
Also @nuclearpinguin -> what helps is to increase memory used by Docker in OSX. I think default 2GB is too little. See https://github.com/apache/airflow/blob/master/BREEZE.rst#prerequisites where I recommend to set it to 4GB

**Comment by potiuk** [comment]
I'd love to merge it :). it's rather simple.

**Review by kaxil — APPROVED** [review]

**Pull request merged**